sentry_streams/sentry_streams/k8s/generate.py

```diff
@@ -0,0 +1,197 @@
+        labels = copy.deepcopy(common_labels)
+        labels["segment"] = str(segid)
+
+        deployment = copy.deepcopy(deployment_template)
+
+        deployment["metadata"]["labels"] = _merge_labels(
+            deployment["metadata"].get("labels", {}),
+            labels,
+        )
+        deployment["spec"]["selector"]["matchLabels"] = _merge_labels(
+            deployment["spec"]["selector"].get("matchLabels", {}),
+            labels,
+        )
+        deployment["spec"]["template"]["metadata"]["labels"] = _merge_labels(
+            deployment["spec"]["template"]["metadata"]["labels"],
+            labels,
+        )
+
+        deployment["spec"]["replicas"] = segment["parallelism"]
+        deployment["metadata"]["name"] = f"{pipeline_name}-{segid}"
+
+        volumes = deployment["spec"]["template"]["spec"].setdefault("volumes", [])
+        volumes.append(cm_volume)
+
+        for container in deployment["spec"]["template"]["spec"]["containers"]:
+            if container["name"] == container_name:
+                container["image"] = image
+
+                env = container.get("env", [])
+                env.append({"name": "SEGMENT_ID", "value": str(segid)})
+                container["env"] = env
+
+                volume_mounts = container.setdefault("volumeMounts", [])
+                volume_mounts.append(cm_volume_mount)
+
+                container["args"] = [
+                    "--config",
+                    cm_volume_mount["mountPath"],
+                ]
+
+        deployments.append(deployment)
+
+    return deployments
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Generate k8s resources from a sentry_streams deployment config."
+    )
+    parser.add_argument(
+        "--config",
+        type=argparse.FileType("r"),
+        help="Path to a deployment config file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--deployment-template",
+        type=argparse.FileType("r"),
+        help="Path to a deployment template file.",
+        default=importlib.resources.files("sentry_streams")
+        .joinpath("k8s/templates/deployment.yaml")
+        .open("r"),
+    )
+    parser.add_argument(
+        "--configmap-template",
+        type=argparse.FileType("r"),
+        help="Path to a configmap template file.",
+        default=importlib.resources.files("sentry_streams")
+        .joinpath("k8s/templates/configmap.yaml")
+        .open("r"),
+    )
+    parser.add_argument(
+        "--output",
+        type=argparse.FileType("w"),
+        help="Output target file. Defaults to stdout.",
+        default=sys.stdout,
+    )
+    parser.add_argument(
+        "--container-name",
+        type=str,
+        help="Streams segment application container name.",
+        default="segment",
+    )
+    parser.add_argument(
+        "--image",
+        type=str,
+        help="Segment container image.",
+        required=True,
+    )
```

Comment:
Would this be a getsentry/snuba image?

Reply:
How would we include sidecars, environment vars and additional configmaps that sentry will need?

Reply:
By using specific deployment template.